test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2336,4 +2337,19 @@ public static <T extends Throwable> T expectThrows(Class<T> expectedType, Reques
             () -> builder.get().decRef() // dec ref if we unexpectedly fail to not leak transport response
         );
     }
+
+    /**
+     * Call a supplier that might throw and fail if it throws, useful to avoid try/catch boilerplate or cumbersome
+     * propagation of checked exceptions around something you assume should never throw.
+     *
+     * @param supplier A supplier that may throw an exception
+     * @return The value returned by the supplier
+     */
+    protected static <T> T assertDoesNotThrow(CheckedSupplier<T, ?> supplier) {
+        try {
+            return supplier.get();
+        } catch (Exception e) {
+            return fail(e);
+        }
+    }
```

Comment:
This might be controversial. As the javadoc suggests it's just to reduce boilerplate in places where we don't want checked exceptions to propagate, and we don't expect them to occur. There's [something like this in JUnit 5](https://junit.org/junit5/docs/5.3.0/api/org/junit/jupiter/api/Assertions.html#assertDoesNotThrow(org.junit.jupiter.api.function.Executable)). I kept the definitions local because I'm not trying to introduce a grand new pattern here, but rather a useful convenience given the new behaviour of `PlainActionFuture#result`.

Reply:
:+1: I'd be inclined to call it `safeGet` since it's kinda doing the same thing as `safeGet(Future<T>)`, getting a result and asserting that this succeeded.

Reply:
Good call